pilot/pkg/networking/core/v1alpha3/fake.go

```diff
@@ -92,7 +97,10 @@ func NewConfigGenTest(t test.Failer, opts TestOptions) *ConfigGenTest {
 	configs := getConfigs(t, opts)
 	configStore := memory.MakeWithLedger(collections.Pilot, &model.DisabledLedger{}, true)
 
-	configController := memory.NewSyncController(configStore)
+	cc := memory.NewSyncController(configStore)
+	controllers := []model.ConfigStoreCache{cc}
+	controllers = append(controllers, opts.ConfigStoreCaches...)
+	configController, _ := configaggregate.MakeWriteableCache(controllers, cc)
 	go configController.Run(stop)
```

Comment:
I think we add event handlers in xds/fake - are we going to miss events if we start this here?

Reply:
So the concern is the things in ConfigStoreCaches (ie the ingress controller) will miss events since we don't have the handlers added yet? That seems to be true and just happens to work for this specific case/test. We may need to do more cleanup here between xds/v1alpha3/the real istiod bootstrapping.

Reply:
It actually looks like that's whats racing in the unit-tests here. Seems like there is quite a bit of bootstrap/ that's not covered by server_test. Reusing that logic here or expanding coverage in server_test would help uncover other cases like this.

Reply:
I sent a fix which fixes the race but its not super clean.